core/domain/acl_decorators.py

```diff
@@ -126,3 +167,320 @@ def test_can_view_stats(self, exploration_id, **kwargs):
             raise self.PageNotFoundException
 
     return test_can_view_stats
+
+
+def can_create_exploration(handler):
+    """Decorator to check whether the user can create an exploration."""
+
+    def test_can_create(self, **kwargs):
+        if (
+                (role_services.ACTION_CREATE_EXPLORATION in self.actions) and
+                user_services.has_fully_registered(self.user_id)):
+            return handler(self, **kwargs)
+        else:
+            raise self.UnauthorizedUserException(
+                'You do not have credentials to create an exploration.')
+
+    return test_can_create
+
+
+def can_create_collection(handler):
+    """Decorator to check whether the user can create a collection."""
+
+    def test_can_create(self, **kwargs):
+        if (
+                (role_services.ACTION_CREATE_COLLECTION in self.actions) and
+                user_services.has_fully_registered(self.user_id)):
+            return handler(self, **kwargs)
+        else:
+            raise self.UnauthorizedUserException(
+                'You do not have credentials to a create collection.')
+
+    return test_can_create
+
+
+def can_access_creator_dashboard(handler):
+    """Decorator to check whether the user can access creator dashboard
+    page.
+    """
+
+    def test_can_access(self, **kwargs):
+        if self.user_id is None:
+            self.redirect(utils.set_url_query_parameter(
+                feconf.SIGNUP_URL, 'return_url',
+                '/notifications_dashboard'))
```

Comment:
From my point of view, raising 401 error is fine instead of redirecting.

Even if the button is not shown, but we have to protect the site for cases when user can change the url. (I think I don't completely understant your point here by `Will the "Create" button on the navbar still work, when the user is logged out`)